src/js/media-controller.js

```diff
@@ -880,6 +942,34 @@ const monitorForMediaStateReceivers = (
   return unsubscribe;
 };
 
+let testMediaEl;
+export const getTestMediaEl = (nodeName = 'video') => {
+  if (testMediaEl) return testMediaEl;
+  testMediaEl = document?.createElement?.(nodeName);
+  return testMediaEl;
+};
+
+export const hasVolumeSupportAsync = async (mediaEl = getTestMediaEl()) => {
+  if (!mediaEl) return false;
+  const prevVolume = mediaEl.volume;
+  mediaEl.volume = prevVolume / 2 + 0.1;
+  return Promise.resolve().then(() => {
+    mediaEl.volume !== prevVolume;
+  });
+};
+
+export const hasPipSupport = (mediaEl = getTestMediaEl()) =>
+  typeof mediaEl?.requestPictureInPicture === 'function';
+
+const pipSupported = hasPipSupport();
+
+let volumeSupported;
+const volumeSupportPromise = hasVolumeSupportAsync().then((supported) => {
+  volumeSupported = supported;
+});
+
+const airplaySupported = !!window.WebKitPlaybackTargetAvailabilityEvent;
```

Comment:
This also depends on the specific media element having the appropriate API, right?

Reply:
In the case of `<youtube-video>` (and similar iframe medias) the platform will support airplay, the media may/will not expose the required `media.webkitShowPlaybackTargetPicker` API. So as implemented, the airplay button will still show itself, but when you click it, you'll [get a console warning today](https://github.com/muxinc/media-chrome/blob/05eee08657a0716c70093be5b3d6a20eb7c8f9f7/src/js/media-controller.js#L224).

We don't need to address that in this PR, but want to make sure we're on the same page as the issue. For a youtube-video + media-chrome use case, I don't think the console warning or just permanently disabling the button is the right answer. However we get there, the button should ultimately be hidden in the same way that it is on platforms that don't support airplay.